File: RestAPI101_Front/flutter_restapi101/lib/models/user/userChangePasswordDTO.dart
import 'package:flutter/cupertino.dart';
import 'package:json_annotation/json_annotation.dart';

part 'userChangePasswordDTO.g.dart';

@immutable
@JsonSerializable()
class UserChangePasswordDTO {
  final String oldPassword;
  final String newPassword;

  UserChangePasswordDTO({required this.oldPassword, required this.newPassword});

  factory UserChangePasswordDTO.fromJson(Map<String, dynamic> json) => _$UserChangePasswordDTOFromJson(json);
  Map<String, dynamic> toJson() => _$UserChangePasswordDTOToJson(this);
}

File: RestAPI101.Backend/RestAPI101_Back/DTOs/User/UserChangePasswordDTO.cs
namespace RestAPI101_Back.DTOs
{
    public class UserChangePasswordDTO
    {
        public string OldPassword { get; }
        public string NewPassword { get; }

        public UserChangePasswordDTO(string oldPassword, string newPassword)
        {
            OldPassword = oldPassword;
            NewPassword = newPassword;
        }
    }
}
